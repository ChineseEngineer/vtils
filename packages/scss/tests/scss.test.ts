import _ from 'shelljs'
import fs from 'fs'
import globby from 'globby'
import path from 'path'
import postcss from 'postcss'

async function processScssFile(file: string) {
  const parsedTestFilePath = path.parse(file)
  const {css} = await postcss([
    require('@csstools/postcss-sass')({
      includePaths: [parsedTestFilePath.dir],
    }),
  ]).process(
    fs.readFileSync(file).toString(),
    {syntax: require('postcss-scss'), from: file},
  )
  return css
}

test('单测正常', async () => {
  const testFiles = globby.sync(
    path.join(__dirname, '../src/**/*.test.scss'),
    {absolute: true},
  )
  await Promise.all(
    testFiles.map(async testFile => {
      expect(
        await processScssFile(testFile),
      ).toMatchSnapshot(
        path.parse(testFile).base,
      )
    }),
  )
})

test('打包正常', async () => {
  _.cd(path.join(__dirname, '..'))
  _.exec('yarn bundle')
  expect(
    await processScssFile(path.join(__dirname, 'bundle.test.scss')),
  ).toMatchSnapshot('bundle.test.scss')
})
