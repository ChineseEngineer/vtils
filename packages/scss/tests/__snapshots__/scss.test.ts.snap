// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`单测正常: _ellipsis.test.scss 1`] = `
".one-line {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.two-lines {
  overflow: hidden;
  /*! autoprefixer: off */
  /* autoprefixer: off */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  /* autoprefixer: on */
  /*! autoprefixer: on */
}

.ten-lines {
  overflow: hidden;
  /*! autoprefixer: off */
  /* autoprefixer: off */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 10;
  /* autoprefixer: on */
  /*! autoprefixer: on */
}"
`;

exports[`单测正常: _maskFeedback.test.scss 1`] = `
".button {
  position: relative;
  cursor: pointer;
  user-select: none;
}
.button:hover::after, .button.hover::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.1);
}
.button:active::after, .button.active::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.2);
}
.button.absolute {
  position: absolute;
  cursor: pointer;
  user-select: none;
}
.button.absolute:hover::after, .button.absolute.hover::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.1);
}
.button.absolute:active::after, .button.absolute.active::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.2);
}"
`;

exports[`单测正常: _opacityFeedback.test.scss 1`] = `
".button {
  cursor: pointer;
  user-select: none;
}
.button:hover, .button.hover {
  opacity: 0.8;
}
.button:active, .button.active {
  opacity: 0.6;
}"
`;

exports[`单测正常: _size.test.scss 1`] = `
".square100 {
  width: 100px;
  height: 100px;
}

.w100-h200 {
  width: 100px;
  height: 200px;
}"
`;

exports[`打包正常: bundle.test.scss 1`] = `
".demo {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  position: relative;
  cursor: pointer;
  user-select: none;
  cursor: pointer;
  user-select: none;
  width: 40px;
  height: 40px;
}
.demo:hover::after, .demo.hover::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: green;
}
.demo:active::after, .demo.active::after {
  content: \\"\\";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: blue;
}
.demo:hover, .demo.hover {
  opacity: 0.1;
}
.demo:active, .demo.active {
  opacity: 0.5;
}"
`;
